File: converteDecBin.c
/**
 * 
 * @autor Maurício Freire
 * Mostra a representação binária de um dado número.
 * Shows the binary representation of a chosen number.
*/
#include <stdio.h>

int main()
{
    int i, n = 170;
    int bin[7];

    printf("%d em binario:\n", n);
    for (i = 15; i >= 0; i--) {
        if (n % 2 == 0) {
            bin[i] = 0;
            n = n / 2;
        } else {
            bin[i] = 1;
            n = n / 2;
        }
    }

    for (i = 0; i < 16; i++) {
        printf("%d", bin[i]);
        if (i % 8 == 0) printf(" ");
    }
    return 0;
}

File: converteDecBin.rb
=begin

  @autor Maurício Freire
  Mostra a representação binária de um dado número.
  Shows the binary representation of a chosen number.
=end

print "Digite um número: "
n = gets.chomp.to_i
bin = []

# Converte o número digitado para 0 e 1.
# Convertes the typed number to 0 and 1.
print "#{n} em binário é: "
for i in 15.downto(0)
    if n % 2 == 0
        bin[i] = 0
    else
        bin[i] = 1
    end
    n = n / 2
end

# Exibe a representação binária do número.
# Shows the binary representation of the number.
for i in 0..15
    if i % 8 == 0
        print " "
    end
    print bin[i]
end
